source/time_tables/thermometer.cpp

```diff
@@ -200,6 +200,8 @@ void Thermometer::generate_thermometer(const std::vector<vector_idx>& sps) {
     boost::range::sort(stop_point_lists, [](const vector_idx& a, const vector_idx& b) { return a.size() > b.size(); });
 
     if (stop_point_lists.size() > 1) {
+        // Clean old results
+        thermometer.clear();
```

Comment:
In term of pattern, to be more robust against changes, I would rather see a new vector to hold the generated stop_point_lists and then move the result like in:

```cpp
void Thermometer::generate_thermometer(const std::vector<vector_idx>& sps) {
   vector_idx new_thermometer;
   [...]
   for (auto v : stop_point_lists) 
       new_thermometer.push_back(v);
   [...]
   thermometer = move(new_thermometer);
```

But I'm also fine with this change :)